tensorflow/python/kernel_tests/nn_ops/pooling_ops_test.py

```diff
@@ -85,6 +85,9 @@ def GetTestConfigsDicts(v1_fn,
     configs1 += [(data_format, use_gpu, dtypes.float16),
                  (data_format, use_gpu, dtypes.float64)]
 
+    if use_gpu:
+      configs1 += [(data_format, use_gpu, dtypes.bfloat16)]
```

Comment:
Can you add gradient tests too? Unfortunately this is tricky because the gradient tests currently only test with fp32 :(. It's fine if you only add bf16 to a single gradient test.

And same with the 3D pooling ops test.

Reply:
Thanks for the review, I added gradient tests for both 2d and 3d pooling.